rpc/src/cache_block_meta_service.rs

```diff
@@ -60,9 +64,42 @@ impl CacheBlockMetaService {
         Self { thread_hdl }
     }
 
-    fn cache_block_meta(bank: &Bank, blockstore: &Blockstore) -> Result<(), BlockstoreError> {
-        blockstore.cache_block_time(bank.slot(), bank.clock().unix_timestamp)?;
-        blockstore.cache_block_height(bank.slot(), bank.block_height())
+    fn cache_block_meta(
+        bank: &Bank,
+        blockstore: &Blockstore,
+        max_complete_rewards_slot: &Arc<AtomicU64>,
+    ) -> Result<(), BlockstoreError> {
+        let slot = bank.slot();
+
+        blockstore.cache_block_time(slot, bank.clock().unix_timestamp)?;
+        blockstore.cache_block_height(slot, bank.block_height())?;
```

Comment:
Ah, the other difference is that writing block times and block heights in the thread will have to wait on writing rewards. I guess this is okay because `rpc::check_blockstore_writes_complete()` checks `max_complete_rewards_slot`? So we shouldn't return blocks with empty block times or heights, even during the rewards period?

Reply:
> Ah, the other difference is that writing block times and block heights in the thread will have to wait on writing rewards

Yeah, I guess so with a few caveats:
- For any given slot, the write order defined by this function is 1. block time 2. block height 3. block rewards. So, in the context of a single slot, writing rewards has to wait for writing block height & time 
- In the context of multiple slots, yes, rewards getting written for a slot could slow down block height & time writes for the next slot if the crossbeam channel is backed up
- That being said, this thread will get a block every 400ms at steady state which is adequate time to perform the writes. If this wasn't sufficient time, we'd see slow memory growth as the channel size grows and eventually RPC queryable block lagging significantly behind the tip of cluster

> I guess this is okay because rpc::check_blockstore_writes_complete() checks max_complete_rewards_slot?

I actually wasn't familiar with this check 😅 . But I think you are correct that this PR works with this check

> So we shouldn't return blocks with empty block times or heights, even during the rewards period?

I also agree with this piece, and arguably, this PR makes things better. We previously had no guarantee that the block height and time were written by the time `rpc::check_blockstore_writes_complete()` returned `Ok(())`. As the blow code calls out, we might have just returned `None` for these fields:
https://github.com/anza-xyz/agave/blob/b778359d9296bd956acf7d0dc1dcc450b7c27d11/ledger/src/blockstore.rs#L2812-L2822